Fix Header order on Source and add compatibility for empty data on SDMX-ML Data messages

javihern98 opened the pull request:
Closes #217 

- Fixed Header order and added test to write and read empty data. 
- Ensure we write a Header with Source and references that is valid against the SDMX-ML 2.1 XSD.

sosna reviewed (approved):
This looks good, thanks @javihern98.

Just one minor comment: This issue should have been caught by schema validation. Should we therefore add tests validating the generated XML output against the SDMX-ML schemas?

javihern98 commented:
> This looks good, thanks @javihern98.
> 
> Just one minor comment: This issue should have been caught by schema validation. Should we therefore add tests validating the generated XML output against the SDMX-ML schemas?

Hi @sosna, we added already a test to use the read_sdmx function with validate=True, in other issues we added it as well, so this should be covered now.
